python-lib/ontology_tagging/formatter_by_document.py

```diff
@@ -0,0 +1,442 @@
+import json
+import numpy as np
+import pandas as pd
+import logging
+from time import perf_counter
+
+from typing import AnyStr
+from typing import List
+from typing import Tuple
+from collections import defaultdict
+
+from spacy.tokens import Span
+from spacy.tokens import Doc
+
+from utils.plugin_io_utils import unique_list
+from utils.plugin_io_utils import generate_unique_columns
+from utils.nlp_utils import get_span_text
+from utils.nlp_utils import unicode_normalize_text
+
+from .ontology_tagging_formatting import FormatterBase
+
+
+class FormatterByDocument(FormatterBase):
+    """Inherits from FormatterBase class.
+    Class to write a dataframe which contains one row per document, with associated tags and keywords stored as array columns.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(FormatterByDocument, self).__init__(*args, **kwargs)
```

Comment:
Nitpick: ditto can only use `super().__init__(*args, **kwargs)` in python3

Reply:
In PR #21